smarts/core/id_actor_capture_manager.py

```diff
@@ -58,17 +59,29 @@ def step(self, sim):
             for a, (b, m) in self._actor_for_agent.items()
             if m.start_time <= sim.elapsed_sim_time and a in social_vehicle_ids
         ):
-            assert isinstance(mission.entry_tactic, IdEntryTactic)
-            patience_expiry = mission.start_time + mission.entry_tactic.patience
-            if sim.elapsed_sim_time > patience_expiry:
-                self._log.error(
+            entry_tactic = mission.entry_tactic
+            assert isinstance(entry_tactic, IdEntryTactic)
+            vehicle = sim.vehicle_index.vehicle_by_id(actor_id)
+            condition_kwargs = ActorCaptureManager._gen_all_condition_kwargs(
+                agent_id,
+                mission,
+                sim,
+                vehicle.state if vehicle is not None else None,
+                entry_tactic.condition.requires,
+            )
+            condition_result = entry_tactic.condition.evaluate(**condition_kwargs)
+            if condition_result == ConditionState.EXPIRED:
+                print(condition_result)
+                self._log.warning(
                     f"Actor aquisition skipped for `{agent_id}` scheduled to start between "
-                    + f"`{mission.start_time}` and `{patience_expiry}` has expired with no vehicle."
-                    f"`simulation time: {sim.elapsed_sim_time}`"
+                    + f"`Condition `{entry_tactic.condition}` has expired with no vehicle."
```

Comment:
Conditions have an `EXPIRED` state which is lowest priority on OR operations and highest on AND operations. So it is still possible to expire. If the condition never generates `EXPIRED` it will never expire.

There are cases like such:

```python console
>>> condition = LiteralCondition(ConditionState.FALSE).expire(20)
>>> condition.evaluate(time=10)
<ConditionState.FALSE: 0>
>>> condition.evaluate(time=21)
<ConditionState.EXPIRED: 2>
```

I have some thoughts that we should add `.then(other_condition)` and/or `.on_hold(count_seconds=5, hold_state=ConditionState.FALSE, emit=ConditionState.EXPIRED)` to handle additional temporal logic.